satpy/readers/ahi_hsd.py

```diff
@@ -357,7 +359,10 @@ def __init__(self, filename, filename_info, filetype_info,
                                                 filetype_info)
 
         self.is_zipped = False
-        self._unzipped = unzip_file(self.filename, prefix=str(filename_info['segment']).zfill(2))
+        if isinstance(self.filename, str):
+            self._unzipped = unzip_file(self.filename, prefix=str(filename_info['segment']).zfill(2))
+        elif isinstance(self.filename, FSFile):
+            self._unzipped = unzip_FSFile(self.filename, prefix=str(filename_info['segment']).zfill(2))
```

Comment:
Could you put this `if/elif` into a `unzip_file` function and then refactor that function to use a `_unzip_local_file` and an `_unzip_fsfile`? There is a lot of duplicate code here so it'd be nice to hide it away in the helper/utility function.